server/app/incident_service.go

```diff
@@ -778,9 +779,54 @@ func (s *IncidentServiceImpl) UpdateStatus(incidentID, userID string, options St
 		}()
 	}
 
+	if options.Status == StatusArchived && incidentToModify.ExportChannelOnArchiveEnabled {
+
+		fileID, err := s.exportFileToChannel(incidentToModify.Name, incidentToModify.OwnerUserID, incidentToModify.ChannelID)
+		if err != nil {
+			_, _ = s.poster.PostMessage(incidentToModify.ChannelID, "Incident Collaboration failed to export channel information. Contact your System Admin for more information.")
+			return errors.Wrap(err, "failed to export file to a channel")
+		}
+
+		if err = s.poster.PostDM(incidentToModify.OwnerUserID, &model.Post{Message: fmt.Sprintf("Incident %s exported succesfully", incidentToModify.Name), FileIds: []string{fileID}}); err != nil {
+			return errors.Wrap(err, "failed to send exported channel result to incident's commander")
+		}
+
+	}
+
 	return nil
 }
 
+func (s *IncidentServiceImpl) exportFileToChannel(incidentName string, ownerUserID string, channelID string) (string, error) {
+	// set url and query string
+	exportPluginURL := fmt.Sprintf("plugins/com.mattermost.plugin-channel-export/api/v1/export?format=csv&channel_id=%s", channelID)
+
+	req, err := http.NewRequest(http.MethodGet, exportPluginURL, bytes.NewBufferString(""))
+	req.Header.Add("Mattermost-User-ID", ownerUserID)
+	if err != nil {
+		errMessage := "failed to create request to generate export file"
+		s.pluginAPI.Log.Warn(errMessage, "plugin", "channel-export", "error", err)
+
+		return "", errors.Wrap(err, errMessage)
+	}
+
+	res := s.pluginAPI.Plugin.HTTP(req)
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusOK {
+		file, err := s.pluginAPI.File.Upload(res.Body, fmt.Sprintf("%s.csv", incidentName), channelID)
+		if err != nil {
+			errMessage := "unable to upload the exported file to a channel"
+			s.pluginAPI.Log.Error(errMessage, "Channel ID", channelID, "Error", err)
+
+			return "", errors.Wrap(err, errMessage)
+		}
+
+		return file.Id, nil
+	}
+
+	return "", errors.New(fmt.Sprintf("There is an error when make a request to upload file with status code %s", strconv.Itoa(res.StatusCode)))
```

Comment:
Maybe return early if there is an error? Something like the following:

```suggestion
	if res.StatusCode != http.StatusOK {
		return "", errors.New(fmt.Sprintf("There is an error when make a request to upload file with status code %s", strconv.Itoa(res.StatusCode)))
	}
	
	file, err := s.pluginAPI.File.Upload(res.Body, fmt.Sprintf("%s.csv", incidentName), channelID)
	if err != nil {
		errMessage := "unable to upload the exported file to a channel"
		s.pluginAPI.Log.Error(errMessage, "Channel ID", channelID, "Error", err)

		return "", errors.Wrap(err, errMessage)
	}

	return file.Id, nil
```

Reply:
Done @agarciamontoro

Reply:
Perfect!